### PR TITLE
Support check_mk agents 1.5+ and systemd based agent configuration

### PR DESCRIPTION
**Pull request by alexjfisher**
This PR supercedes https://github.com/voxpupuli/puppet-check_mk/pull/9.
It rebases the commit from that PR and adds back the agent configuration options.  Both xinetd and systemd based configuration are supported.

### CODE REVIEW

File: manifests/agent.pp [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
-  $server_dir   = '/usr/bin',
-  $use_cache    = false,
-  $user         = 'root',
-  $version      = undef,
-  $workspace    = '/root/check_mk',
-  $package      = undef,
-  $mrpe_checks  = {},
-) {
-  validate_hash($mrpe_checks)
+  Optional[String] $filestore = undef,
```
Comment (bastelfreak) [inline manifests/agent.pp]: datatypes \o/

Comment (bastelfreak) [reply manifests/agent.pp]: You probably don't want to spend more time here and enforce minimal string length / datatypes within the arrays?

File: manifests/agent/params.pp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Class: check_mk::agent::params
+#
+# @summary Sets the defaults for the agent class.
+# @api private
+#
+class check_mk::agent::params {
+    case $::osfamily {
```
Comment (bastelfreak) [inline manifests/agent/params.pp]:
```suggestion
    case $facts['os']['family'] {
```

File: manifests/agent/params.pp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    'RedHat': {
+      $config_dir = '/etc/check-mk-agent'
+    }
+    'Debian': {
+      $config_dir = '/etc/check_mk'
+    }
+    default: {
+      fail("OS familly ${::osfamily} is not managed!")
```
Comment (bastelfreak) [inline manifests/agent/params.pp]:
```suggestion
      fail("OS familly ${facts['os']['family']} is not managed!")
```

File: manifests/agent/service.pp [hunk 4]
```diff
@@ -1,17 +1,49 @@
...
+#
+class check_mk::agent::service (
+  Boolean $use_xinetd = $check_mk::agent::use_xinetd,
+)
+{
+  if $use_xinetd {
+    ensure_packages(['xinetd'])
+    Package['xinetd'] ~> Service['xinetd']
+
+    service { 'check_mk.socket':
```
Comment (bastelfreak) [inline manifests/agent/service.pp]: did you consider not deploying the systemd config if xinted is used? I don't have an opinion here, just curious.

File: metadata.json [hunk 5]
```diff
@@ -31,13 +31,22 @@
       "operatingsystem": "Debian",
       "operatingsystemrelease": [
         "6",
```
Comment (bastelfreak) [inline metadata.json]: we should purge debian 6 and 7 because they are EOL.

File: templates/agent/check_mk.service-drop-in.epp [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<%- |
```
Comment (bastelfreak) [inline templates/agent/check_mk.service-drop-in.epp]: please add a first line like `# THIS FILE IS MANAGED BY PUPPET`

File: templates/agent/encryption.cfg.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+ENCRYPTED=yes
```
Comment (bastelfreak) [inline templates/agent/encryption.cfg.erb]: why did you choose an erb template here?

File: metadata.json (hunk 5 above)
Comment (alexjfisher) [reply metadata.json]: Yes.  I'd like to put out one release with them still in and purge them immediately after.

File: manifests/agent/service.pp (hunk 4 above)
Comment (alexjfisher) [reply manifests/agent/service.pp]: The package installs this systemd file (even on systems without systemd).  This module manages the drop-in files.

File: manifests/agent/params.pp (hunk 3 above)
Comment (alexjfisher) [reply manifests/agent/params.pp]: OK.  I'll update `check_mk::params` as well as `check_mk::agent::params` (and fix the spelling of `family`!)

File: templates/agent/encryption.cfg.erb (hunk 7 above)
Comment (alexjfisher) [reply templates/agent/encryption.cfg.erb]: I'll change this in my next PR (which will add some acceptance tests for this parameter)

File: manifests/agent/params.pp (hunk 3 above)
Comment (alexjfisher) [reply manifests/agent/params.pp]: Fixed

File: templates/agent/check_mk.service-drop-in.epp (hunk 6 above)
Comment (alexjfisher) [reply templates/agent/check_mk.service-drop-in.epp]: Fixed

File: manifests/agent/params.pp (hunk 2 above)
Comment (alexjfisher) [reply manifests/agent/params.pp]: Fixed

**Comment by zyronix** [comment]
Great work Alex en Tim! Thanks! Looking forward for the first release of this module!

**Pull request merged**